pkg/cloud/ovirt/clients/machineservice.go

```diff
@@ -344,26 +355,70 @@ func  (is *InstanceService) FindVirtualMachineIP(id string,excludeAddr map[strin
 	for _, reportedDevice := range reportedDeviceSlice.Slice() {
 		nicName, _ := reportedDevice.Name()
 		if !nicRegex.MatchString(nicName) {
-			klog.Infof("ovirt vm id: %s ,  skipped nic %s , naming regex mismatch",id, nicName)
+			klog.Infof("ovirt vm id: %s ,  skipped nic %s , naming regex mismatch", id, nicName)
 			continue
 		}
 
 		ips, hasIps := reportedDevice.Ips()
 		if hasIps {
 			for _, ip := range ips.Slice() {
-				ipres, hasAddress := ip.Address()
+				ipAddress, hasAddress := ip.Address()
 
-				if _, ok := excludeAddr[ipres]; ok {
-					klog.Infof("address %s is excluded from usable IPs", ipres)
+				if _, ok := excludeAddr[ipAddress]; ok {
+					klog.Infof("ipAddress %s is excluded from usable IPs", ipAddress)
 					continue
 				}
 
 				if hasAddress {
-					klog.Infof("ovirt vm id: %s , found usable IP %s", id, ipres)
-					return ipres,nil
+					klog.Infof("ovirt vm id: %s , found usable IP %s", id, ipAddress)
+					return ipAddress, nil
 				}
 			}
 		}
 	}
 	return "", fmt.Errorf("coudlnt find usable IP address for vm id: %s", id)
 }
+
+func (is *InstanceService) getAffinityGroups(cID string, agNames []string) (ag []*ovirtsdk.AffinityGroup, err error) {
+	var ags []*ovirtsdk.AffinityGroup
+	res, err := is.Connection.SystemService().ClustersService().
+		ClusterService(cID).AffinityGroupsService().
+		List().Send()
+	if err != nil {
+		return nil, err
+	}
+	agNamesMap := make(map[string]*ovirtsdk.AffinityGroup)
+	for _, af := range res.MustGroups().Slice() {
+		agNamesMap[af.MustName()] = af
+	}
+	for _, agName := range agNames {
+		if _, ok := agNamesMap[agName]; ok {
+			ags = append(ags, agNamesMap[agName])
+		}
```

Comment:
So if the AG was not found we just silently ignore it or am I getting this wrong?

Reply:
Yep that used to be the logic to ignore errors with adding the VM to the Affinity group, but we agreed to actually fail in case we can't add the VM to the affinity group so we should do the same in case an affinity group that has been specified doesn't exist.

I'll add a clear error message for that

Reply:
Done